sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -923,7 +966,11 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
     // Make sure a plan's subtree does not contain outer references
     def failOnOuterReferenceInSubTree(p: LogicalPlan): Unit = {
       if (hasOuterReferences(p)) {
-        failAnalysis(s"Accessing outer query column is not allowed in:\n$p")
+        throw new AnalysisException(
+          errorClass = "INVALID_SUBQUERY_EXPRESSION",
+          errorSubClass = "AGGREGATE_FUNCTION_MIXED_OUTER_LOCAL_REFERENCES",
+          origin = p.origin,
+          messageParameters = Array.empty[String])
```

Comment:
Should we include the query plan here? Otherwise, it's hard to tell what's wrong.

Reply:
Same as about, we include the `origin` here which indicates the location in the input SQL query at which the error took place. Let's talk about it offline though, we can go either way on this.